app/src/main/java/com/android/slangify/ui/activities/CreateChallengeActivity.java

```diff
@@ -74,66 +75,50 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
                     Manifest.permission.RECORD_AUDIO,
                     Manifest.permission.CAMERA,
                     Manifest.permission.WRITE_EXTERNAL_STORAGE);
-
         }
 
         mLanguageAutoComplete.setThreshold(1);
 
-        LanguagesRepository languagesRepository = new LanguagesRepository();
-        languagesRepository.getLanguages(new IRepositoryCallback<LanguageModel>() {
-            @Override
-            public void onSuccess(ArrayList<LanguageModel> result) {
-                //save data
-                languageModels = result;
+        languageModels = SharedPreferencesUtils.getLanguagesFromCache(CreateChallengeActivity.this);
 
-                if (languageModels != null && languageModels.size() > 0) {
-                    mLanguageAutoComplete.setHint(languageModels.get(0).getName());
-                }
-
-                String[] mStringArray = new String[languageModels.size()];
-                for (int i = 0; i < languageModels.size(); i++) {
-                    mStringArray[i] = languageModels.get(i).getName();
-                }
+        if (languageModels != null && languageModels.size() > 0) {
+            mLanguageAutoComplete.setHint(languageModels.get(0).getName());
+        }
```

Comment:
what about the "else" part? say the languages didn't have enough time/connection to load yet.

Reply:
There is no else part prior the change as well. :/
I didn't change that, it just we will download it to SP in an earlier step